crates/rbuilder/src/provider/ipc_state_provider.rs

```diff
@@ -0,0 +1,580 @@
+use std::{
+    fmt::Debug,
+    path::{Path, PathBuf},
+    sync::Arc,
+    time::Duration,
+};
+
+use alloy_consensus::{constants::KECCAK_EMPTY, Header};
+use alloy_eips::{BlockId, BlockNumHash, BlockNumberOrTag};
+use alloy_primitives::{BlockHash, BlockNumber, StorageKey, StorageValue, U64};
+use dashmap::DashMap;
+use quick_cache::sync::Cache;
```

Comment:
Seems like both dashmap and quickcache are used to handle some caches, is there a reason to use different ones?